### PR TITLE
app-misc/terminal-colors: Bump to version 3.0.2

### PR DESCRIPTION
**Pull request by ThamognyaKodi**
- Bump to versions 3.0.2
- Added Latest Manifests
- Added my self as maintainer
- This ebuild uses the packaging on my server (just like the previous ebuilds which uses radhermit distfiles, but are not maintained) due to the flaw of packaging upstream.

Signed-off-by: Thamognya Kodi <contact@thamognya.com>

### CODE REVIEW

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
+PYTHON_REQ_USE="ncurses"
+
+inherit python-r1
+
+DESCRIPTION="A tool to display color charts for 8, 16, 88, and 256 color terminals"
+HOMEPAGE="http://zhar.net/projects/shell/terminal-colors"
+SRC_URI="https://assets.thamognya.com/gentoo/distfiles/${P}.tar.gz"
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:12]: Where'd you get the source code? Why not use the original URL? Or is this your own fork (should be advertised in the HOMEPAGE then)

**Review by juippis — COMMENTED** [review]
There's no need to push 3.0.1 as when you push 3.0.2 with the same keywords, no one will ever install 3.0.1.

File: app-misc/terminal-colors/metadata.xml [hunk 2]
```diff
@@ -1,5 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-<!-- maintainer-needed -->
+  <maintainer type="person" proxied="yes">
+    <name>Thamognya Kodi</name>
+    <email>contact@thamognya.com</email>
+    <description>Primary maintainer</description>
```
Comment (juippis) [inline app-misc/terminal-colors/metadata.xml:7]: I don't think we do this anymore. Did you find this from some guide?

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild (hunk 1 above)
Comment (ThamognyaKodi) [reply app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:12]: Ah, it is not a fork rather it continues what radhermit did which was to package the script seperately in the previous ebuilds. But unfortunately he or she does not maintain does distfiles any longer hence I had to make the tar.gz. I am currently trying to work on making a ebuild that won't rely on that.

File: app-misc/terminal-colors/metadata.xml (hunk 2 above)
Comment (ThamognyaKodi) [reply app-misc/terminal-colors/metadata.xml:7]: I have seen it on https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Taking_over_an_existing_package . I do not know if it outdated.

File: app-misc/terminal-colors/Manifest [hunk 3]
```diff
@@ -1 +1,2 @@
 DIST terminal-colors-2.3.gz 5424 BLAKE2B e08635c0e989a430805159829b133f18b7c30959c945e4373a77442ac15628bb8bf3a5a2d14cd4f5705a5dd332297fa3d41611b52f56baaf9b1a882c26626714 SHA512 64fbfe766563c4c25bec23d1657187414f97b6f92e8a4a35337563210ef9873db495376676ef0a2f4e440ad77a6c1b43ee9c27fe04b34c2839eb4fffd5289a00
+DIST terminal-colors-3.0.2.tar.gz 5511 BLAKE2B e0d80997d1b95c88d00ec322a79519efce6f531ac66dc09462ada53eccdb6e3a0f150f6325d708475c1b50c3976f88291f24b500b32d350513712bd22624a061 SHA512 e00a9c6b3e16bb88cfa78f0fad76ce192641f693424aaf7c65d87dcf46324d3650cd553cdd53dae26e91d36dd57104cd27a173375771b79cc540e1cb4246258e
```
Comment (juippis) [inline app-misc/terminal-colors/Manifest:2]:
```
!!! Fetched file: terminal-colors-3.0.2.tar.gz VERIFY FAILED!
!!! Reason: Filesize does not match recorded size
!!! Got:      152226
!!! Expected: 5511
```
That's a huge size difference. Please remove your local distfile, try again, and make sure this is the file that works with your ebuild as intended.

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~x64-macos"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}"
+
+src_install() {
+	mv "${S}/${P}/${PN}" "${S}" || die "moving the script failed"
+	rm -rf "${S}/${P}" || die "rm failed"
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:24]: I'm sorry but.. what is going on in here? Can you explain what you're doing and why.

Comment (ThamognyaKodi) [reply app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:24]: The line 23, moves the script from the opened folder to the work directory, and in line 24, it deletes the opened folder, which contains the README, license and other stuff that are not necessary for the package.

Comment (juippis) [reply app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:24]: Okay, I think I see what's happening here. The previous distfile just provided the final terminal-colors file, while the real upstream one provides much more, including a setup.py. You're already calling `python_foreach_impl python_newscript ${PN} ${PN}` to just install what you want. But for that to work, I guess we'll have to...

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
+PYTHON_REQ_USE="ncurses"
+
+inherit python-r1
+
+DESCRIPTION="A tool to display color charts for 8, 16, 88, and 256 color terminals"
+HOMEPAGE="http://zhar.net/projects/shell/terminal-colors"
+SRC_URI="https://github.com/eikenb/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}"
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:13]:
```suggestion
```

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~x64-macos"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}"
+
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:21]:
```
RDEPEND="${PYTHON_DEPS}"
BDEPEND="${PYTHON_DEPS}"
```

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+src_install() {
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:22]:
```suggestion

src_compile() { :; }

src_install() {
```

File: app-misc/terminal-colors/terminal-colors-3.0.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+	mv "${S}/${P}/${PN}" "${S}" || die "moving the script failed"
+	rm -rf "${S}/${P}" || die "rm failed"
+	python_foreach_impl python_newscript ${PN} ${PN}
+}
```
Comment (juippis) [inline app-misc/terminal-colors/terminal-colors-3.0.2.ebuild:26]:
```suggestion
src_install() {
	python_foreach_impl python_newscript ${PN} ${PN}
}
```

Although I personally find the README file useful, so you can add `einstalldocs` call to your `Src_install`.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**